src/Datadog.Trace.ClrProfiler.Managed/AutoInstrumentation/Msmq/IMessageQueue.cs

```diff
@@ -0,0 +1,146 @@
+using System;
+
+namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.Msmq
+{
+    /// <summary>
+    /// message queue proxy
+    /// </summary>
+    public interface IMessageQueue
+    {
+        /// <summary>
+        ///     gets a value indicating whether a cache of connections will be maintained
+        ///     by the application.
+        /// Returns:
+        ///   true to create and use a connection cache; otherwise, false.
+        /// </summary>
+        bool EnableConnectionCache { get; }
```

Comment:
I don't think all of these properties are used in the integration? We have to generate code for each extra property, so AFAIK it's best to only specify the properties you actually need